cdc/generation.cc

```diff
@@ -54,32 +54,37 @@ namespace cdc {
 extern const api::timestamp_clock::duration generation_leeway =
     std::chrono::duration_cast<api::timestamp_clock::duration>(std::chrono::seconds(5));
 
-stream_id::stream_id()
-    : _first(-1), _second(-1) {}
-
 stream_id::stream_id(int64_t first, int64_t second)
-    : _first(first)
-    , _second(second) {}
+    : _value(bytes::initialized_later(), 2 * sizeof(int64_t))
+{
+    std::copy_n(reinterpret_cast<int8_t*>(&first), sizeof(int64_t), _value.begin());
+    std::copy_n(reinterpret_cast<int8_t*>(&second), sizeof(int64_t), _value.begin() + sizeof(int64_t));
+}
 
 bool stream_id::is_set() const {
-    return _first != -1 || _second != -1;
+    return !_value.empty();
 }
 
 bool stream_id::operator==(const stream_id& o) const {
-    return _first == o._first && _second == o._second;
+    return _value == o._value;
 }
 
 int64_t stream_id::first() const {
-    return _first;
+    assert(_value.size() == 2 * sizeof(int64_t));
+    int64_t res;
+    std::copy_n(_value.begin(), sizeof(int64_t), reinterpret_cast<int8_t *>(&res));
+    return res;
 }
 
 int64_t stream_id::second() const {
-    return _second;
+    assert(_value.size() == 2 * sizeof(int64_t));
+    int64_t res;
+    std::copy_n(_value.begin() + sizeof(int64_t), sizeof(int64_t), reinterpret_cast<int8_t *>(&res));
+    return net::ntoh(res);
```

Comment:
`second` method performs endianness conversion, but `first` does not. Is it intended/correct?

Shouldn't we also byte-swap in the constructor `stream_id::stream_id(int64_t first, int64_t second)`?

Reply:
That's a leftover of my previous idea. Good catch. Thanks - will fix.